src/roiextractors/extractors/inscopixextractors/inscopixsegmentationextractor.py

```diff
@@ -0,0 +1,99 @@
+"""Inscopix Segmentation Extractor."""
+
+from typing import Optional
+import platform
+import numpy as np
+
+from ...extraction_tools import PathType, ArrayType
+from ...segmentationextractor import SegmentationExtractor
+
+
+class InscopixSegmentationExtractor(SegmentationExtractor):
+    """A segmentation extractor for Inscopix."""
+
+    extractor_name = "InscopixSegmentationExtractor"
+    installed = True
+    is_writable = False
+    mode = "file"
+    installation_mesg = ""
+
+    def __init__(self, file_path: PathType):
+        """Initialize a InscopixSegmentationExtractor instance.
+
+        Main class for extracting segmentation data from Inscopix format.
+
+        Parameters
+        ----------
+        file_path: str
+            The location of the folder containing Inscopix *.mat output file.
+        """
+        if platform.system() == "Darwin":
+            raise ImportError(
+                "The isx package is currently not natively supported on macOS with Apple Silicon. "
+                "Installation instructions can be found at: "
+                "https://github.com/inscopix/pyisx?tab=readme-ov-file#install"
+            )
+
+        import isx
+
+        SegmentationExtractor.__init__(self)
+        self.file_path = file_path
+        self.cell_set = isx.CellSet.read(file_path)
+
+    def get_num_rois(self) -> int:
+        return self.cell_set.num_cells
+
+    def get_roi_image_masks(self, roi_ids=None) -> np.ndarray:
+        if roi_ids is None:
+            roi_idx_ = range(self.get_num_rois())
+        else:
+            all_ids = self.get_roi_ids()
+            roi_idx_ = [all_ids.index(i) for i in roi_ids]
+
+        masks = [self.cell_set.get_cell_image_data(roi_id) for roi_id in roi_idx_]
+        if len(masks) == 1:
+            return masks[0]
+        return np.stack(masks)
+
+    def get_roi_ids(self) -> list:
+        return [self.cell_set.get_cell_name(x) for x in range(self.get_num_rois())]
+
+    def get_image_size(self) -> ArrayType:
+        try:
+            # Swap dimensions to return (width, height)
+            pixels = self.cell_set.spacing.num_pixels
+            return (pixels[1], pixels[0])
+        except AttributeError:
+            if self.get_num_rois() > 0:
+                shape = self.cell_set.get_cell_image_data(0).shape
+                # Swap dimensions to return (width, height)
+                return (shape[1], shape[0])
+            return None
```

Comment:
Instead of returning None here, I think we should throw an error. Is num_rois=0 even a sensible situation for an inscopix segmentation?